pkg/network/multus/nad.go

```diff
@@ -42,3 +50,26 @@ func NetAttachDefNamespacedName(namespace, fullNetworkName string) types.Namespa
 		Name:      fullNetworkName,
 	}
 }
+
+func GetNetworkToResourceMap(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) (networkToResourceMap map[string]string, err error) {
+	networkToResourceMap = make(map[string]string)
+	for _, network := range vmi.Spec.Networks {
+		if network.Multus != nil {
+			nadNamespacedName := NetAttachDefNamespacedName(vmi.Namespace, network.Multus.NetworkName)
+			crd, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(nadNamespacedName.Namespace).Get(context.Background(), nadNamespacedName.Name, metav1.GetOptions{})
+			if err != nil {
+				return map[string]string{}, fmt.Errorf("failed to locate network attachment definition %s", nadNamespacedName.String())
+			}
+			networkToResourceMap[network.Name] = GetResourceNameForNetwork(crd)
+		}
+	}
+	return
+}
+
+func GetResourceNameForNetwork(network *networkv1.NetworkAttachmentDefinition) string {
```

Comment:
What's the reason for exporting this function?

Reply:
The `nad_test.go` file is in the `multus_test` package.
It wouldn't be able to test this function without exporting it.

Reply:
That is not a strong enough reason to expose the function.
A unit is not necessarily defined as being a function, therefore I would expect the functionality to be covered by exercising enough scenarios through the function that call it.

Is there a technical reason that makes it hard to do so?

Originally I commented here also about it being redundant, but when I saw the last comment, I moved it there.

Reply:
I've tried instantiating a fake client implementing the `kubecli.KubevirtClient` interface.
I've also tried changing the client's type to `networkattachmentdefinitionclient.Interface` - it breaks lots of unit tests since they use gomock and don't expect their client to call `t.virtClient.NetworkClient()`.

Additionally, the unit tests of `getResourceNameForNetwork` were removed.